regression/cbmc/array_of_bool_as_bitvec/main.c

```diff
@@ -0,0 +1,20 @@
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+
+__CPROVER_bool x[8];
+
+void main()
+{
+  bool y[8];
+  bool *z = malloc(8 * sizeof(bool));
+
+  memset(y, 0, sizeof(y));
+  memset(z, 0, sizeof(8 * sizeof(bool)));
```

Comment:
Wouldn't
```
bool y[8] = { 0 };
bool *z = calloc(8, sizeof(boo));
```
also do the trick, or does this not trigger the desired code paths? Likewise, is it actually necessary to use `bool`, or could `_Bool` or `int` equally be used?

Reply:
Only `__CPROVER_bool` seems to trigger the desired code paths.
I have these `bool` variables (which are actually encoded as `int`) just to make sure their encoding is unaffected by the proposed changes.

I am not sure what `_Bool` is -- I will try this out and also modify the array initializations as you suggested.

Reply:
Looks like `_Bool` doesn't trigger the code path either.

I have now implemented the changes as suggested.